### ISSUE TITLE
Discussion: GPL?

### ISSUE THREAD

**Issue by jpursell** [d-e-s-o/apca#45]
Just wanted to comment to the author: GPL seems like an extreme choice for a stock trading library as opposed to something like the MIT license. It's my understanding that anyone using this library is legally obligated to release their source code to the public. Is that incorrect?

**Comment by All8Up** [comment]
I admit to having an absolute hatred of GPL3 ingrained in me from 30+ years of commercial software work where the lawyers all had the same reaction: "Don't touch GPL3, no exceptions.".  I'm not doing anything commercial with APCA, probably why I never thought to check the license, but it does limit what I can/will open source that is related to it.  Specifically I have a number of libraries I use which are MIT/ZLib that I'd have to post if I opened up my backtesting, data warehouse, etc but if I did so with linkage to APCA given this license and the anti-Tivo clauses, I'd have to change those licenses.  That's something I won't do so yeah, the framework I've been writing will stay private and never see the light of day.  Unfortunate.

Not my choice obviously but I would definitely throw in my $0.02 towards a change to just about any license other than GPL3.  Even downgrading to GPL2 would be preferable because very specifically the anti-Tivo clauses are despicable IMO.
